tests/tokens_service_test.go

```diff
@@ -0,0 +1,101 @@
+package tests
+
+import (
+	"context"
+	"testing"
+	"time"
+
+	"github.com/google/go-cmp/cmp"
+	"github.com/google/go-cmp/cmp/cmpopts"
+	"github.com/google/uuid"
+
+	"github.com/flow-hydraulics/flow-wallet-api/jobs"
+	"github.com/flow-hydraulics/flow-wallet-api/tests/internal/test"
+	"github.com/flow-hydraulics/flow-wallet-api/transactions"
+)
+
+func Test_TokensSetup(t *testing.T) {
+	cfg := test.LoadConfig(t, testConfigPath)
+	svc := test.GetServices(t, cfg).GetTokens()
+
+	_, testAccount, err := test.GetServices(t, cfg).GetAccounts().Create(context.Background(), true)
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	type input struct {
+		sync      bool
+		tokenName string
+		address   string
+	}
+	type expect struct {
+		job   *jobs.Job
+		tx    *transactions.Transaction
+		error bool
+	}
+	testCases := []struct {
+		name   string
+		input  input
+		expect expect
+	}{
+		{
+			name: "success",
+			input: input{
+				sync:      false,
+				tokenName: "flowToken",
+				address:   testAccount.Address,
+			},
+			expect: expect{
+				job: &jobs.Job{
+					Type:                   transactions.TransactionJobType,
+					State:                  jobs.Init,
+					Error:                  "",
+					Result:                 "",
+					ExecCount:              int(1),
+					ShouldSendNotification: true,
+				},
+				tx: &transactions.Transaction{
+					TransactionType: transactions.FtSetup,
+					ProposerAddress: testAccount.Address,
+				},
+				error: false,
+			},
+		},
+		{
+			name: "fail address not found",
+			input: input{
+				sync:      false,
+				tokenName: "flowToken",
+				address:   "0x0ae53cb6e3f42a79",
+			},
+			expect: expect{
+				job:   nil,
+				tx:    nil,
+				error: true,
```

Comment:
It's preferred to specify what error is expected to ensure expected behaviour. The `Setup` function could be improved to return the `record not found` error early (instead of the current `error while getting new transaction: error while building transaction: error while getting user authorizer: WHERE conditions required`) :)

Reply:
fixed `Setup` function to return `record not found` early 👍